package/kedro_viz/server.py

```diff
@@ -398,70 +414,114 @@ def format_pipeline_data(
 
         for data_set in node.outputs:
-            namespace = data_set.split("@")[0]
-            namespace_to_layer[namespace] = dataset_to_layer.get(data_set)
-            namespace_id = _hash(namespace)
+            dataset_namespace = data_set.split("@")[0]
+            if node.namespace:
+                namespace_to_modular_pipelines[dataset_namespace] = _expand_namespaces(
+                    node.namespace
+                )
+            dataset_namespace_to_layer[dataset_namespace] = dataset_to_layer.get(
+                data_set
+            )
+            namespace_id = _hash(dataset_namespace)
             edge = {"source": task_id, "target": namespace_id}
             if edge not in edges_list:
                 edges_list.append(edge)
-            namespace_tags[namespace].update(node.tags)
+            dataset_namespace_tags[dataset_namespace].update(node.tags)
             node_dependencies[task_id].add(namespace_id)
+
     # Parameters and data
-    for namespace, tag_names in sorted(namespace_tags.items()):
-        is_param = _is_namespace_param(namespace)
-        node_id = _hash(namespace)
+    for dataset_namespace, tag_names in sorted(dataset_namespace_tags.items()):
+        is_param = _is_namespace_param(dataset_namespace)
+        node_id = _hash(dataset_namespace)
 
         _JSON_NODES[node_id] = {
             "type": "parameters" if is_param else "data",
-            "obj": _get_dataset_data_params(namespace),
+            "obj": _get_dataset_data_params(dataset_namespace),
         }
-        if is_param and namespace != "parameters":
+        if is_param and dataset_namespace != "parameters":
             # Add "parameter_name" key only for "params:" prefix.
-            _JSON_NODES[node_id]["parameter_name"] = namespace.replace("params:", "")
+            _JSON_NODES[node_id]["parameter_name"] = dataset_namespace.replace(
+                "params:", ""
+            )
 
         if node_id not in nodes:
             nodes[node_id] = {
                 "type": "parameters" if is_param else "data",
                 "id": node_id,
-                "name": _pretty_name(namespace),
-                "full_name": namespace,
+                "name": _pretty_name(dataset_namespace),
+                "full_name": dataset_namespace,
                 "tags": sorted(tag_names),
-                "layer": namespace_to_layer[namespace],
+                "layer": dataset_namespace_to_layer[dataset_namespace],
                 "pipelines": [pipeline_key],
+                "modular_pipelines": namespace_to_modular_pipelines.get(
+                    dataset_namespace, []
+                ),
             }
             nodes_list.append(nodes[node_id])
         else:
             nodes[node_id]["pipelines"].append(pipeline_key)
+            mod_pipelines = namespace_to_modular_pipelines.get(dataset_namespace, [])
+            check = nodes[node_id]["modular_pipelines"] and all(
```

Comment:
Sorry, what is this checking?

Reply:
These are lists so the check is verifying whether there are any modular pipelines and if so that the ones on the dataset don't already exist in the list.